src/DssSpell.t.base.sol

```diff
@@ -1999,4 +2001,38 @@ contract DssSpellTestBase is Config, DssTest {
         vat.move(address(this), address(0x0), vat.dai(address(this)));
     }
 
+    // Test checks that any contracts in the chainlog which are a ward of
+    //  the pause proxy are also have the ESM authed. Any new contracts
+    //  that are added to the chainlog should have an ESM auth if the
+    //  pause proxy is authorized on them. If it is not necessary to remove
+    //  proxy access after a shutdown event, it can be added to the
+    //  test/esm_exceptions.sol array. Add a comment to indicate why this
+    //  authorization is not necessary.
+    function _checkESMWards() internal {
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        bytes32[] memory _tags       = chainLog.list();
+        bytes32[] memory _exceptions = exceptions.list();
+
+        for (uint256 i = 0; i < _tags.length; i++) {
+            bool _found = false;
+            for (uint256 j = 0; j < _exceptions.length; j++) {
+                if (_tags[i] == _exceptions[j]) {
+                    _found = true;
```

Comment:
Consider to adding `break` here to avoid unnecessary iterations over the `_exceptions` list for elements that have already been found.
```suggestion
                    _found = true;
                    break;
```